pulp_rpm/app/tasks/publishing.py

```diff
@@ -95,21 +92,46 @@ def prepare_metadata_files(self, contents, folder=None):
 
         return repomdrecords
 
-    def get_packages(self, contents):
+    def publish_artifacts(self, contents):
         """
-        Get packages from content.
+        Publish artifacts.
 
         Args:
             contents (pulpcore.plugin.models.Content): A list of contents.
 
-        Returns:
-            packages (pulp_rpm.models.Package): A list of packages.
+        """
+        published_artifacts = []
+        for content_artifact in ContentArtifact.objects.filter(content__in=contents).iterator():
```

Comment:
instead of going to each kind of content and prefetch its `content_artifact`, it was better to use `ContentArtifact` for all contents, using `iterator`, so it would not cache all the data in memory